torchrl/envs/gym_like.py

```diff
@@ -100,6 +132,42 @@ def _output_transform(self, step_outputs_tuple: Tuple) -> Tuple:
             )
         return step_outputs_tuple
 
+    def set_info_dict_reader(self, info_dict_reader: callable) -> GymLikeEnv:
+        """
+        Sets an info_dict_reader function. This function should take as input an
+        info_dict dictionary and the tensordict returned by the step function, and
+        write values in an ad-hoc manner from one to the other.
+
+        Args:
+            info_dict_reader (callable): a callable taking a input dictionary and
+                output tensordict as arguments. This function should modify the
+                tensordict in-place.
+
+        Returns: the same environment with the dict_reader registered.
+
+        Examples:
+            >>> from torchrl.envs import GymWrapper, default_info_dict_reader
+            >>> reader = default_info_dict_reader(["my_info_key"])
+            >>> # assuming "some_env-v0" returns a dict with a key "my_info_key"
+            >>> env = GymWrapper(gym.make("some_env-v0")).set_info_dict_reader(info_dict_reader=reader)
+            >>> tensordict = env.reset()
+            >>> tensordict = env.rand_step(tensordict)
+            >>> assert "my_info_key" in tensordict.keys()
+
+        """
+        self.info_dict_reader = info_dict_reader
+        return self
+
+    @property
+    def info_dict_reader(self):
+        if "_info_dict_reader" not in self.__dir__():
```

Comment:
I assume that this is checking if `_info_dict_reader` attribute exists or not. I am not sure if checking the `self.__dir__()` is the recommended method. I have seen use of `hasattr` method lot more. So just flagging this, in case it turns out to be relevant.

Reply:
Interesting. I didnt realise hashattr would do that but it makes sense!

Reply:
Yeah it's not a clever function
It just runs getattr and tells you ok if it did not return an error...